tests/bdd/test_ana_validate_nexus_swap.py

```diff
@@ -0,0 +1,221 @@
+    """the only I/O path degrades."""
+    pass
+
+
+@then("fio client should successfully complete with the replaced I/O path")
+def fio_client_should_successfully_complete_with_the_replaced_io_path(
+    fio_completes_successfully,
+):
+    """fio client should successfully complete with the replaced I/O path."""
+    pass
+
+
+@then(
+    "it should be possible to create a second nexus and connect it as the second path"
+)
+def it_should_be_possible_to_create_a_second_nexus_and_connect_it_as_the_second_path(
+    publish_to_node_2, connect_to_node_2
+):
+    """it should be possible to create a second nexus and connect it as the second path."""
+    pass
+
+
+@then("it should be possible to remove the first failed I/O path")
+def it_should_be_possible_to_remove_the_first_failed_io_path(remove_first_path):
+    """it should be possible to remove the first failed I/O path."""
+    pass
+
+
+"""" FixTure Implementations """
+
+
+@pytest.fixture
+def background():
+    Deployer.start_with_args(
+        [
+            "-j",
+            "-m=3",
+            "-w=10s",
+            "--cache-period=1s",
+            "--mayastor-env=NEXUS_NVMF_ANA_ENABLE=1,NEXUS_NVMF_RESV_ENABLE=1",
+            "--agents-env=TEST_NEXUS_NVMF_ANA_ENABLE=1",
+        ]
+    )
+    ApiClient.pools_api().put_node_pool(
+        POOL_NODE, POOL_UUID, CreatePoolBody(["malloc:///disk?size_mb=100"])
+    )
+    ApiClient.volumes_api().put_volume(
+        VOLUME_UUID, CreateVolumeBody(VolumePolicy(False), 1, VOLUME_SIZE)
+    )
+    volume = ApiClient.volumes_api().put_volume_target(
+        VOLUME_UUID, TARGET_NODE_1, Protocol("nvmf")
+    )
+    yield volume
+    Deployer.stop()
+
+
+@pytest.fixture
+def connect_to_first_path(background):
+    volume = background
+    device_uri = volume.state["target"]["deviceUri"]
+    yield nvme_connect(device_uri)
+    nvme_disconnect(device_uri)
+
+
+@pytest.fixture
+def run_fio_to_first_path(connect_to_first_path):
+    device = connect_to_first_path
+    desc = nvme_list_subsystems(device)
+    assert (
+        len(desc["Subsystems"]) == 1
+    ), "Must be exactly one NVMe subsystem for target nexus"
+    subsystem = desc["Subsystems"][0]
+    assert len(subsystem["Paths"]) == 1, "Must be exactly one I/O path to target nexus"
+    assert subsystem["Paths"][0]["State"] == "live", "I/O path is not healthy"
+    # Launch fio in background and let it always run along with the test.
+    fio = Fio("job", "randread", device, runtime=FIO_RUNTIME).build()
+    return subprocess.Popen(fio, shell=True)
+
+
+@pytest.fixture
+def degrade_first_path():
+    Docker.kill_container(TARGET_NODE_1)
+
+
+@pytest.fixture
+def publish_to_node_2(background):
+    volume = background
+    device_uri = volume.state["target"]["deviceUri"]
+
+    try:
+        ApiClient.volumes_api().del_volume_target(VOLUME_UUID)
+    except ApiException as e:
+        # Time out or node not online
+        assert e.status == 408 or e.status == 412
```

Comment:
Maybe we can define constants for the error codes so that it reads a little easier.

Reply:
Done